service/src/test/java/org/databiosphere/workspacedataservice/service/RecordOrchestratorServiceTest.java

```diff
@@ -127,6 +130,63 @@ void deleteRecordType() {
         "describeRecordType should have thrown an error");
   }
 
+  @Test
+  void deleteAttribute() {
+    // Arrange
+    setUpDeleteAttributeTest();
+
+    // Act
+    recordOrchestratorService.deleteAttribute(INSTANCE, VERSION, TEST_TYPE, "attr2");
+
+    // Assert
+    assertAttributes(Set.of("id", "attr1"));
+  }
+
+  @Test
+  void deletePrimaryKeyAttribute() {
+    // Arrange
+    setUpDeleteAttributeTest();
+
+    // Act/Assert
+    assertThrows(
```

Comment:
When it's possible for an exception of the caught type (`DeleteAttributeRequestException`) to be thrown for multiple reasons, I think it's a good idea to also assert that the message (and therefore the reason) matches what you're expecting.

I think you can use this approach.
```suggestion
    var thrown = assertThrows(...)
    assertThat(var.getMessage()).contains("Unable to delete ID");
```

Reply:
Oops. Misunderstood the "message" parameter of `assertThrows` here. Done in https://github.com/DataBiosphere/terra-workspace-data-service/pull/451/commits/64ce7dd0a5b1e6ab8b42b2925e53348d2c8aa173.